rir/src/utils/CodeHandle.h

```diff
@@ -28,6 +28,7 @@ class CodeHandle {
         code->skiplistLength = skiplistLength(sourceSize);
         code->callSiteLength = callSiteLength;
         code->perfCounter = 0;
+        code->isFormalPromise = isFormal;
```

Comment:
I don't really like the name "formal promise". It is kind of confusing. can we have something more meaningful, like "(compiled) default arguments", or such.